script/script_hooks/destinations

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+if [ -z "$DEFAULT_OSX_DESTINATION" ]; then
```

Comment:
@guidomb at the end I moved most of the env variable declaration to the `install` script as suggested. It will generate some defaults for Simulator Name & OS version for iOS, and for OSX it will use the current arch (no value at all).

This variables will honor the values defined in the current bash context so stuff like this will work
```bash
IOS_DESTINATION_VERSION="9.0" IOS_DESTINATION_SIMULATOR_NAME="iPhone 6s Plus" script/cibuild
```
So now we can customise where it should run the tests on-the-fly and while keeping the values in the `.env` file generated on install.

This made me tweak what platform Carthage was going to build during bootstrap and also how the script pick the schemes to build.
The former is just an extra variable `CARTHAGE_BUILD_PLATFORM` in the `.env` file with a comma separated list of them, e.g. "Mac,iOS" or "All" (The values listed in `carthage help build`).
The latter is just checking if the variable `SCHEME` is defined in which case the script will build only that scheme otherwise the default behaviour will be used (which is the default since that variable is not stored in the env file)

There is only one issue I when you update existing scripts and this variable is not defined cause the .env script is already generated maybe is not a big deal but it will break without them. Should we define some defaults in the script or improve the "update" flow?